### DISCUSSION TITLE
Lesson 4: Fund() function is not working

### DISCUSSION THREAD

**Discussion by komi786** [Cyfrin/foundry-full-course-cu#444] [Q&A] [answered]
I don't seem to figure out the issue in the code. There is no syntax error and I hope no logical error as much as I could check. I have researched online and checked the discussion section for relevant issue occurrences but didn't find a solution.
I am able to successfully deploy fundMe.sol on sepolia testnet on metamask wallet  where I have 0.6 Ether. However, when I try to execute Fund function, it returns the following error:
For value, I gave bigger than 5$ in wei like 4000000000000000 wei or 0,04 eth

**"transact to FundMe.addFund errored: Returned error: {"jsonrpc":"2.0","error":"execution reverted","id":7150564985197293}".**
I have attached my code for your reference. 

**FundMe.sol**
`//SPDX-License-Identifier:MIT

pragma solidity ^0.8.18;

import {PriceConvertor} from "./PriceConvertor.sol";

contract FundMe{

    using PriceConvertor for uint256;
    address[] public  founders;
    mapping (address founders => uint256 amountFounded) public addresstoAmountFounded;
    uint256 public  minimumUSD = 5e18;
    function addFund() public payable {

        require(msg.value.getconversionRate() >= minimumUSD , "Didn't send enough Eth");
        founders.push(msg.sender);  
        addresstoAmountFounded[msg.sender] = addresstoAmountFounded[msg.sender] + msg.value;  //add what they have previously founded

    }    
}`

 **PriceConvertor.sol**
`// SPDX-License-Identifier: MIT
pragma solidity 0.8.18;

import {AggregatorV3Interface} from  "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

library PriceConvertor{

    function getPrice() public view  returns (uint256) {
        
        AggregatorV3Interface pricefeed =  AggregatorV3Interface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419);
        (, int answer,, , ) = pricefeed.latestRoundData();
        return uint256(answer * 1e10);

     }
    function getconversionRate(uint256 ethamount) public view returns(uint256){

        uint256 ethprice = getPrice();
        uint256 ethAmountinUSD = (ethprice * ethamount) / 1e18;
        return  ethAmountinUSD;

    }
     function getVersion() public view  returns (uint256){

         uint version = AggregatorV3Interface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419).version();
         return version ;
     }

}`

**Comment by TSHEPO-CLOUD** [comment, marked as answer]
Good day,
Try getting all your functions in your library to be `internal` instead of `public`
And also don't forget the price feed pairs contract address for ETH/USD on sepolia teatnet `0x694AA1769357215DE4FAC081bf1f309aDC325306`  which must be included in the `AggregatorV3Interface`

**Reply by komi786** [reply]
Thank you @TSHEPO-CLOUD
